src/util/compareThought.ts

```diff
@@ -221,3 +223,21 @@ export const compareThoughtByUpdated: ComparatorFunction<Thought> = (a: Thought,
 /** Compare two thoughts by their lastUpdated timestamp in descending order (newest first). Fall back to compareReasonable if created at the same time. */
 export const compareThoughtByUpdatedDescending: ComparatorFunction<Thought> = (a: Thought, b: Thought) =>
   compare(b.lastUpdated, a.lastUpdated) || compareReasonable(a.value, b.value)
+
+/** Get note value or fallback for a Thought or string input. */
+const getNoteOrFallback = (state: State, input: Thought | string): string => {
+  if (typeof input === 'string') return input
+  return noteValue(state, input.id) || input.rank.toString()
```

Comment:
You can't compare numbers lexicographically, because `'2' > '10'`.

Take advantage of [makeOrderedComparator](https://github.com/cybersemics/em/blob/92d6d2d02373b1a20ec7ae3003b8e0dc1deb2b97/src/util/compareThought.ts#L137) or follow the [compareDateAndOther](https://github.com/cybersemics/em/blob/92d6d2d02373b1a20ec7ae3003b8e0dc1deb2b97/src/util/compareThought.ts#L130) pattern instead.